Functions/Assertions/HaveParameter.Tests.ps1

```diff
@@ -2,87 +2,92 @@ Set-StrictMode -Version Latest
 
 InModuleScope Pester {
 
-    if ($PSVersionTable.PSVersion.Major -ge 5) {
-        function Invoke-DummyFunction {
-            param(
-                [Parameter(Mandatory = $true)]
-                $MandatoryParam,
-
-                [ValidateNotNullOrEmpty()]
-                [DateTime]$ParamWithNotNullOrEmptyValidation = (Get-Date),
-
-                [Parameter()]
-                [ValidateScript(
-                    {
-                        if (-not (Test-Path $_)) {
-                            $errorItem = [System.Management.Automation.ErrorRecord]::new(
-                                ([System.ArgumentException]"Path not found"),
-                                'ParameterValue.FileNotFound',
-                                [System.Management.Automation.ErrorCategory]::ObjectNotFound,
-                                $_
-                            )
-                            $errorItem.ErrorDetails = "Invalid path '$_'."
-                            $PSCmdlet.ThrowTerminatingError($errorItem)
+    $script:functionsBlock = {
```

Comment:
Why is this necessary? The functions seem to be in scope and you don't need to import them again in `BeforeAll`. Are you changing this to make the code cleaner? Because this change is really hard to review and unless it is totally necessary I'd skip it.

Reply:
Without the dot sourcing inside the Describe block, `Get-Command` complains that it cannot find the functions

Reply:
@nohwnd Logs from the build showing the issue: [HaveParameter tests](https://nohwnd.visualstudio.com/Pester/_build/results?buildId=449&view=logs&jobId=986454e2-4a53-5b4f-19e5-426b1a182c9d&taskId=22f2c122-d6a2-5866-e1bb-4f38fae86508&lineStart=2825&lineEnd=2826&colStart=1&colEnd=1)

Reply:
@nohwnd It may be related to [this issue](https://github.com/PowerShell/PowerShell/issues/8022) in PowerShell. Since the blocks discovery and execution is now moved to the internal *Pester.Runtime* module, `Get-Command` seems to be impacted

Reply:
@nohwnd  How do you want to proceed here? The workaround mentioned there is what I was already doing, dot-sourcing the functions so they area reachable. Another alternative would be to define the functions in the global scope

Reply:
Yeah true, I forgot. Just keep it :)